lib/assessment.js

```diff
@@ -39,6 +55,14 @@ var select = function select(array, callback, thisArg) {
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
 var find = function find(array, callback, thisArg) {
+  var result = true;
+  array.forEach(function(currentValue, index, array) {
+
+  if(result = callback(currentValue, index, array)){
+      return currentValue;
```

Comment:
Here's some interesting trouble. `find` won't work as intended for a a reason that will help you deepen your understanding of functions.

When you `return` from a function, the flow of control (i.e., the order code gets executed in) returns to the place immediately following where the function was called. In this case, your callback to `Array.forEach` is being called by `Array.forEach`, so it returns to `Array.forEach` with the value. `Array.forEach` doesn't do anything with the value and just keeps iterating through the array. `Array.forEach` also doesn't return a value, so any values to returned by the callback you give it are simply lost.

As a result, `find` always returns `undefined`. For this question, a classic `for` loop containing the `if` statement you constructed would have worked. 3/4 credit.